lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/stack/StackPresenter.java

```diff
@@ -191,6 +192,11 @@ public void onChildDestroyed(ViewController<?> child) {
         componentLeftButtons.remove(child.getView());
     }
 
+    public void attachNewViewController(ViewController<?> previousVC, ViewController<?> newVC) {
+        Options options = newVC.resolveCurrentOptions(defaultOptions);
+        StatusBarPresenter.instance.onSelectViewController(options.statusBar, previousVC);
```

Comment:
Perhaps this belongs in the `ComponentViewPresenter` (as with the push/pop animations flow)